src/KeyDB.php

```diff
@@ -151,15 +152,15 @@ public function addKey(array $keyMeta)
      *     The key metadata matching $keyId. See self::addKey() and the TUF
      *     specification for the array structure.
      *
-     * @throws \Exception
+     * @throws \InvalidArgumentException
      *     Thrown if the key ID is not found in the keydb database.
      *
      * @see https://github.com/theupdateframework/specification/blob/master/tuf-spec.md#4-document-formats
      */
     public function getKey(string $keyId)
     {
         if (empty($this->keys[$keyId])) {
-            throw new \Exception("Unknown key ID: $keyId");
+            throw new \InvalidArgumentException("Unknown key ID: $keyId");
```

Comment:
Hmm, actually I'm not sure about this one. Key IDs are often read from other TUF repository metadata, so I sort of expect that a malformed remote TUF repository that referenced a nonexistent key would end up throwing here. To my mind that's in the class of bad input problems, not programming problems.

Reply:
Likewise with the unknown role name. What about a `TufException` subclass for all malformed TUF repository issues?

Reply:
I ended up using the pre-existing `\Tuf\Exception\NotFoundException`, which seems to be the right fit for these errors.